ext/crypto/import_key.rs

```diff
@@ -657,11 +673,103 @@ fn import_key_rsaoaep(
   }
 }
 
+fn decode_b64url_to_gen_array<T: ArrayLength<u8>>(
+  b64: &str,
+) -> Result<GenericArray<u8, T>, deno_core::anyhow::Error> {
+  jwt_b64_int_or_err!(val, b64, "invalid b64 coordenate");
+
+  let mut bytes: GenericArray<u8, T> = GenericArray::default();
+  bytes[..val.as_bytes().len()].copy_from_slice(val.as_bytes());
+
+  Ok(bytes)
+}
+
+fn import_key_ec_jwk_to_point(
+  x: String,
+  y: String,
+  named_curve: EcNamedCurve,
+) -> Result<Vec<u8>, deno_core::anyhow::Error> {
+  let point_bytes = match named_curve {
+    EcNamedCurve::P256 => {
+      let x = decode_b64url_to_gen_array(&x)?;
+      let y = decode_b64url_to_gen_array(&y)?;
+
+      p256::EncodedPoint::from_affine_coordinates(&x, &y, false).to_bytes()
+    }
+    EcNamedCurve::P384 => {
+      let x = decode_b64url_to_gen_array(&x)?;
+      let y = decode_b64url_to_gen_array(&y)?;
+
+      p384::EncodedPoint::from_affine_coordinates(&x, &y, false).to_bytes()
+    }
+  };
+
+  Ok(point_bytes.to_vec())
+}
+
+fn import_key_ec_jwk(
+  key_data: KeyData,
+  named_curve: EcNamedCurve,
+) -> Result<ImportKeyResult, deno_core::anyhow::Error> {
+  match key_data {
+    KeyData::JwkPublicEc { x, y } => {
+      let point_bytes = import_key_ec_jwk_to_point(x, y, named_curve)?;
+
+      Ok(ImportKeyResult::Ec {
+        raw_data: RawKeyData::Public(point_bytes.to_vec().into()),
+      })
+    }
+    KeyData::JwkPrivateEc { d, x, y } => {
+      let point_bytes = import_key_ec_jwk_to_point(x, y, named_curve)?;
+
+      let d = decode_b64url_to_gen_array::<U32>(&d)?;
+
+      let secret_key_der = match named_curve {
+        EcNamedCurve::P256 => {
+          let secret_key = p256::SecretKey::from_bytes(&d)?;
+
+          secret_key.to_pkcs8_der().unwrap()
+        }
+        //@todo(sean) - build p384 secret key from jwk, when crate implements to_pkcs8_der
+        /*EcNamedCurve::P384 => {
+          let secret_key = p384::SecretKey::from_bytes(&d)?;
+
+          secret_key.to_pkcs8_der().unwrap()
+        }*/
+        _ => {
+          return Err(data_error("Unsupported namedCurve"));
+        }
+      };
+
+      let oid =
+        <p256::NistP256 as p256::elliptic_curve::AlgorithmParameters>::OID;
+
+      let pki = p256::pkcs8::PrivateKeyInfo::new(
+        p256::pkcs8::AlgorithmIdentifier {
+          oid,
+          parameters: None,
+        },
+        secret_key_der.as_ref(),
+      );
+
+      let pki = p256::pkcs8::PrivateKeyInfo {
+        public_key: Some(&point_bytes),
+        ..pki
+      };
```

Comment:
This is specific to p256.

Reply:
yes. can clean-up when p384 support implemented, maybe using generic impl. as per work in progress on Rust Crypto [here](https://github.com/RustCrypto/elliptic-curves/issues/240)